From 6ea3c0aba4e8b70344a134224ac29f18063a0e61 Mon Sep 17 00:00:00 2001
From: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date: Tue, 31 May 2022 12:28:20 -0300
Subject: [PATCH 123/127] socket: Use 64 bit stat for isfdtype (BZ# 29209)

This is a missing spot initially from 52a5fe70a2c77935.

Checked on i686-linux-gnu.

(cherry picked from commit 87f1ec12e79a3895b33801fa816884f0d24ae7ef)
---
 NEWS                     | 1 +
 sysdeps/posix/isfdtype.c | 4 ++--
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/NEWS b/NEWS
index 4757dea7ab..c943938656 100644
--- a/NEWS
+++ b/NEWS
@@ -36,6 +36,7 @@ The following bugs are resolved with this release:
   [29204] libc: getusershell is not 2038 aware
   [29207] libc: posix_fallocate fallback implementation is not y2038
   [29208] libc: fpathconf(_PC_ASYNC_IO) is not y2038 aware
+  [29209] libc: isfdtype is not y2038 aware
 
 
 Version 2.35
diff --git a/sysdeps/posix/isfdtype.c b/sysdeps/posix/isfdtype.c
index 192c7f9be6..d26c14259e 100644
--- a/sysdeps/posix/isfdtype.c
+++ b/sysdeps/posix/isfdtype.c
@@ -24,12 +24,12 @@
 int
 isfdtype (int fildes, int fdtype)
 {
-  struct stat64 st;
+  struct __stat64_t64 st;
   int result;
 
   {
     int save_error = errno;
-    result = __fstat64 (fildes, &st);
+    result = __fstat64_time64 (fildes, &st);
     __set_errno (save_error);
   }
 
-- 
2.36.1

